src/NuGet.Clients/PackageManagement.UI/Services/NuGetLockService.cs

```diff
@@ -18,8 +20,44 @@ public sealed class NuGetLockService : INuGetLockService, IDisposable
     {
         private readonly SemaphoreSlim _semaphore = new SemaphoreSlim(1, 1);
```

Comment:
Changing this to `AsyncSemaphore` would let you remove the extra code you've written as nested types of this class and still use the `using` style for entering and exiting the semaphore.
Something to consider when you're not in escrow, I guess.